cognite_toolkit/cdf_tk/load.py

```diff
@@ -1189,67 +1252,85 @@ def drop_load_resources(
     nr_of_batches = len(items)
     nr_of_items = sum(len(item) if isinstance(item, Sized) else 1 for item in items)
     if nr_of_items == 0:
-        return
-    nr_of_deleted = 0
-    nr_of_created = 0
-    if load:
+        return DeployResult(name=loader.display_name, created=0, deleted=0, skipped=0, total=0)
+    if action == "deploy":
         print(f"[bold]Uploading {nr_of_items} {loader.display_name} in {nr_of_batches} batches to CDF...[/]")
     else:
         print(f"[bold]Cleaning {nr_of_items} {loader.display_name} in {nr_of_batches} batches to CDF...[/]")
     batches = [item if isinstance(item, Sized) else [item] for item in items]
-    if drop and loader.support_drop and load:
+    if drop and loader.support_drop and action == "deploy":
         print(f"  --drop is specified, will delete existing {loader.display_name} before uploading.")
+
+    # Deleting resources.
+    nr_of_deleted = 0
     if (drop and loader.support_drop) or clean:
         for batch in batches:
-            drop_items: list = []
-            for item in batch:
-                # Set the context info for this CDF project
-                if hasattr(item, "data_set_id") and ToolGlobals.data_set_id is not None:
-                    item.data_set_id = ToolGlobals.data_set_id
```

Comment:
I removed this as we should no longer set data set implicit. If you agree then I will add a check to the build step that complains if `data_set_external_id` is not set for resources that have a `data_set_id` property.

Reply:
Agreed, with one exception that is not really an exception: we need to allow datasetid to be set as data in the APM_Config configuration for infield. I'm not sure what the property is named, I assume it will not be caught by this either way.

Reply:
I agree